dateparser/utils.py

```diff
@@ -86,3 +87,37 @@ def find_date_separator(format):
     m = re.search(r'(?:(?:%[dbBmaA])(\W))+', format)
     if m:
         return m.group(1)
+
+
+class Registry(object):
+
+    _global_dict = {}
+    _tag = 'from_registry'
+
+    def __new__(cls, *args, **kwargs):
+        key = cls.get_key(*args, **kwargs)
+
+        if key in cls._global_dict:
+            return cls._tag_instance(cls._global_dict[key])
+
+        return cls._global_dict.setdefault(
+            key,
+            object.__new__(cls, *args, **kwargs)
+        )
+
+    @classmethod
+    def _tag_instance(cls, instance):
+        setattr(instance, cls._tag, True)
+        return instance
+
+    @classmethod
+    def get_key(cls, *args, **kwargs):
+        raise NotImplemented
+
+    @classmethod
+    def skip_init_if_instance_from_registry(cls, f):
```

Comment:
I'm starting to think if instead of base class it would be more convenient to use class decorator :)